language-server/src/main/java/org/sonarlint/languageserver/SonarLintLanguageServer.java

```diff
@@ -345,6 +342,7 @@ private void updateBinding(@Nullable Map<?, ?> connectedModeProject) {
       logger.error(ClientLogger.ErrorType.START_CONNECTED_ENGINE_FAILED);
       return;
     }
+    telemetry.usedConnectedMode(serverInfo.serverUrl);
```

Comment:
Why not call this after successful binding is confirmed, a few lines below? That would be a bit more accurate.

Reply:
ok